File: cart/src/main/scala/de/leanovate/dose/cart/model/CartItem.scala
package de.leanovate.dose.cart.model

case class CartItem(cartId: String,
                    position: Option[Long],
                    productId: String,
                    productOption: String,
                    amount: Int,
                    product: Option[ActiveProduct],
                    priceInCent: Option[Int]) {
  def fillProduct(_product: ActiveProduct) = {
    copy(
      product = Some(_product),
      priceInCent = _product.options.find(_.name == productOption).map(_.priceInCent * amount))
  }
}

File: billing/src/main/java/de/leanovate/dose/billing/model/CartItem.java
package de.leanovate.dose.billing.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties(ignoreUnknown = true)
public class CartItem {
    @JsonProperty
    public String cartId;

    @JsonProperty
    public int position;

    @JsonProperty
    public String productId;

    @JsonProperty
    public String productOption;

    @JsonProperty
    public int amount;

    @JsonProperty
    public int priceInCent;

    @Override
    public String toString() {

        return "CartItem{" +
                "cartId='" + cartId + '\'' +
                ", position=" + position +
                ", productId='" + productId + '\'' +
                ", productOption='" + productOption + '\'' +
                ", amount=" + amount +
                ", priceInCent=" + priceInCent +
                '}';
    }
}
